### PR TITLE
abstain from registering incomplete services

### PR DESCRIPTION
**Pull request by xabbuh**
fixes symfony/symfony#30948

#FOSSHackathons #EUFOSSA

### CODE REVIEW

**Comment by xabbuh** [comment]
ping @dmaicher @dunglas

**Comment by dmaicher** [comment]
Can confirm that this fixes the issue.

But it also means (without having `symfony/property-info`) I cannot enable the new doctrine validator mapping anymore so it actually works? Or am I missing something?

**Comment by xabbuh** [comment]
something else seems to be wrong, @dmaicher and me are investigating

**Comment by xabbuh** [comment]
In fact, this was not an issue on `1.10.x` at all. I picked the code that already prevent this from happen into `master`.

This is ready to be reviewed.

**Comment by dmaicher** [comment]
@alcaeus somehow this 

https://github.com/doctrine/DoctrineBundle/blob/1.10.x/DependencyInjection/DoctrineExtension.php#L358

was lost during some merge up into master?

File: DependencyInjection/DoctrineExtension.php [hunk 1]
```diff
@@ -355,11 +355,17 @@ protected function ormLoad(array $config, ContainerBuilder $container)
 
         $config['entity_managers'] = $this->fixManagersAutoMappings($config['entity_managers'], $container->getParameter('kernel.bundles'));
 
+        $loadPropertyInfoExtractor = interface_exists('Symfony\Component\PropertyInfo\PropertyInfoExtractorInterface')
```
Comment (alcaeus) [inline DependencyInjection/DoctrineExtension.php]: Can you refactor this to use the `::class` constants?

**Comment by alcaeus** [comment]
This was inadvertently removed in #892. Thanks @xabbuh and @dmaicher for digging into this. I'll see what I can do to prevent this in the future.

**Comment by alcaeus** [comment]
Thanks @xabbuh!

**Comment by xabbuh** [comment]
Thank you for fixing the class.

**Pull request merged**